syntax = "proto3";

enum event_type_enum{

    DEPOSIT = 0;
    WITHDRAW = 1;
    QUERY = 2;
}

message branchEventRequest{
    int32  event_id = 1;
    int32 customer_id = 2;
    event_type_enum event_type = 3;
    double money = 4;
    bool is_propogate = 5;

}

message branchEventResponse{
    int32 event_id = 1;
    event_type_enum event_type = 2;
    double money = 3;
    double balance = 4;
    bool is_success = 5;
    int32 write_id = 6;
}

service branchEventSender{
     rpc MsgDelivery (branchEventRequest) returns (branchEventResponse) {}
}